File: Backend/EduHubLibrary/Domain/Tools/UserProfile.cs
using System;
using System.Collections.Generic;
using EnsureThat;

namespace EduHubLibrary.Domain.Tools
{
    public class UserProfile
    {
        public UserProfile(string name, string email, bool isTeacher, string avatarLink)
        {
            Email = email;
            Name = name;
            IsTeacher = isTeacher;
            AvatarLink = Ensure.Any.IsNotNull(avatarLink);
            Contacts = new List<string>();
        }

        public UserProfile(string name, string email, string aboutUser,
            DateTimeOffset birthYear, Gender gender, bool isTeacher, string avatarLink, List<string> contacts)
        {
            Name = name;
            Email = email;
            AboutUser = aboutUser;
            BirthYear = birthYear;
            Gender = gender;
            IsTeacher = isTeacher;
            AvatarLink = avatarLink;
            Contacts = contacts;
        }

        public string Name { get; internal set; }
        public string Email { get; internal set; }
        public string AboutUser { get; internal set; }
        public DateTimeOffset BirthYear { get; internal set; }
        public Gender Gender { get; internal set; }
        public bool IsTeacher { get; internal set; }
        public string AvatarLink { get; internal set; }
        public List<string> Contacts { get; internal set; }
    }
}

File: Android/EduHub/app/src/main/java/com/example/user/eduhub/Models/UserProfile/UserProfile.java
package com.example.user.eduhub.Models.UserProfile;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.List;

/**
 * Created by User on 30.01.2018.
 */

public class UserProfile implements Serializable {

    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("email")
    @Expose
    private String email;
    @SerializedName("aboutUser")
    @Expose
    private String aboutUser;
    @SerializedName("birthYear")
    @Expose
    private String birthYear;
    @SerializedName("isMan")
    @Expose
    private Boolean isMan;
    @SerializedName("isTeacher")
    @Expose
    private Boolean isTeacher;
    @SerializedName("avatarLink")
    @Expose
    private String avatarLink;
    @SerializedName("contacts")
    @Expose
    private List<String> contacts = null;
    @SerializedName("id")
    @Expose
    private String id;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getAboutUser() {
        return aboutUser;
    }

    public void setAboutUser(String aboutUser) {
        this.aboutUser = aboutUser;
    }

    public String getBirthYear() {
        return birthYear;
    }

    public void setBirthYear(String birthYear) {
        this.birthYear = birthYear;
    }

    public Boolean getIsMan() {
        return isMan;
    }

    public void setIsMan(Boolean isMan) {
        this.isMan = isMan;
    }

    public Boolean getIsTeacher() {
        return isTeacher;
    }

    public void setIsTeacher(Boolean isTeacher) {
        this.isTeacher = isTeacher;
    }

    public String getAvatarLink() {
        return avatarLink;
    }

    public void setAvatarLink(String avatarLink) {
        this.avatarLink = avatarLink;
    }

    public List<String> getContacts() {
        return contacts;
    }

    public void setContacts(List<String> contacts) {
        this.contacts = contacts;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }
}
